### PR TITLE
Increase readability of the CMake build systems code

### PR DESCRIPTION
**Pull request by faisal-bhuiyan**
Quick follow up to the work performed in #225 -- no functional changes here. Goal was to
- simplify the logic as appropriate
- improve readability and maintainability of the CMake build systems code
- add some annotation to the code to provide some context on intention

### CODE REVIEW

**Review by ddement — APPROVED** [review]
Everything looks good to me.

File: cmake/StaticAnalyzers.cmake [hunk 1]
```diff
@@ -1,100 +1,79 @@
...
-    if(NOT
-       CMAKE_CXX_COMPILER_ID
-       MATCHES
-       ".*Clang")
-
+    if(NOT CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
       get_target_property(TARGET_PCH ${target} INTERFACE_PRECOMPILE_HEADERS)
-
-      if("${TARGET_PCH}" STREQUAL "TARGET_PCH-NOTFOUND")
+      if(NOT TARGET_PCH)
```
Comment (ddement) [inline cmake/StaticAnalyzers.cmake:46]: Did you test that this change works like the previous version?  I based things like this off of a template that's been tested on a bunch of platforms, since we don't have an easy way to test these type of edge cases ourselves.

**Review by ddement — COMMENTED** [review]
Everything looks alright from a visual inspection.  Did you change any behavior or was this mostly cosmetic?  If there are behavior changes, I want to give them a quick try.

File: cmake/StaticAnalyzers.cmake (hunk 1 above)
Comment (faisal-bhuiyan) [reply cmake/StaticAnalyzers.cmake:46]: I did not test it, the checks looked a little roundabout way of doing things. Anyways, reverted this change.

**Comment by faisal-bhuiyan** [comment]
> Everything looks alright from a visual inspection. Did you change any behavior or was this mostly cosmetic? If there are behavior changes, I want to give them a quick try.

I don't expect any behavior changes in this PR, but let's make sure. I am able to build on macOS and Linux without the static analyzer checks.

**Comment by ddement** [comment]
At the very least, CppCheck isn't getting set up properly.  It is throwing syntax errors (which should be suppressed), and isn't issuing the proper error code, so they're still treated as warnings.  I saw this locally and you can see it in CI.  Both of these things function properly on main.  Since main currently doesn't pass ClangTidy, I can't test that.

**Comment by faisal-bhuiyan** [comment]
> At the very least, CppCheck isn't getting set up properly. It is throwing syntax errors (which should be suppressed), and isn't issuing the proper error code, so they're still treated as warnings. I saw this locally and you can see it in CI. Both of these things function properly on main. Since main currently doesn't pass ClangTidy, I can't test that.

Fixed the above item.

**Comment by ddement** [comment]
> > At the very least, CppCheck isn't getting set up properly. It is throwing syntax errors (which should be suppressed), and isn't issuing the proper error code, so they're still treated as warnings. I saw this locally and you can see it in CI. Both of these things function properly on main. Since main currently doesn't pass ClangTidy, I can't test that.
> 
> Fixed the above item.

It seems to work for me locally.  Your changes seem to work for my CUDA setup, too.  Looks good to go.
